test/e2e/serial/tests/configuration.go

```diff
@@ -538,19 +539,25 @@ var _ = Describe("[serial][disruptive][slow] numaresources configuration managem
 				e2efixture.Skipf(fxt, "more than one NodeGroup not yet supported, found %d", len(nroOperObj.Spec.NodeGroups))
 			}
 
-			Expect(len(nroOperObj.Status.MachineConfigPools)).To(Equal(len(nroOperObj.Spec.NodeGroups)),
-				"MCP Status mismatch: found %d, expected %d",
-				len(nroOperObj.Status.MachineConfigPools), len(nroOperObj.Spec.NodeGroups),
-			)
-
-			// normalize config to handle unspecified defaults
-			specConf := merge.NodeGroupConfig(
-				nropv1alpha1.DefaultNodeGroupConfig(),
-				*nroOperObj.Spec.NodeGroups[0].Config,
-			)
-			statusConf := nroOperObj.Status.MachineConfigPools[0].Config // shortcut
-			Expect(statusConf).To(Equal(specConf))
+			Eventually(func(g Gomega) {
+				// getting the same object twice is awkward, but still it seems better better than skipping inside a Eventually.
+				err := fxt.Client.Get(context.TODO(), nroKey, &nroOperObj)
+				g.Expect(err).ToNot(HaveOccurred(), "cannot get %q in the cluster", nroKey.String())
+				g.Expect(len(nroOperObj.Status.MachineConfigPools)).To(Equal(len(nroOperObj.Spec.NodeGroups)),
+					"MCP Status mismatch: found %d, expected %d",
+					len(nroOperObj.Status.MachineConfigPools), len(nroOperObj.Spec.NodeGroups),
+				)
+
+				// normalize config to handle unspecified defaults
+				specConf := nropv1alpha1.DefaultNodeGroupConfig()
+				if nroOperObj.Spec.NodeGroups[0].Config != nil {
+					specConf = merge.NodeGroupConfig(specConf, *nroOperObj.Spec.NodeGroups[0].Config)
+				}
 
+				statusConf := nroOperObj.Status.MachineConfigPools[0].Config // shortcut
+				g.Expect(statusConf).ToNot(BeNil())
+				g.Expect(statusConf).To(Equal(specConf))
```

Comment:
I should mention explicitely in a comment that the status must report the default config even if the spec is missing.